src/hotspot/share/memory/metaspace/binList.hpp

```diff
@@ -73,32 +74,25 @@ namespace metaspace {
 // This structure is a bit expensive in memory costs (we pay one pointer per managed
 // block size) so we only use it for a small number of sizes.
 
-template <size_t smallest_word_size, int num_lists>
+template <int num_lists>
 class BinListImpl {
 
   struct Block {
     Block* const _next;
-    const size_t _word_size;
-    Block(Block* next, size_t word_size) :
-      _next(next),
-      _word_size(word_size)
-    {}
+    Block(Block* next) : _next(next) {}
   };
 
-#define BLOCK_FORMAT          "Block @" PTR_FORMAT ": size: " SIZE_FORMAT ", next: " PTR_FORMAT
-#define BLOCK_FORMAT_ARGS(b)  p2i(b), (b)->_word_size, p2i((b)->_next)
+#define BLOCK_FORMAT              "Block @" PTR_FORMAT ": size: " SIZE_FORMAT ", next: " PTR_FORMAT
+#define BLOCK_FORMAT_ARGS(b, sz)  p2i(b), (sz), p2i((b)->_next)
 
-  // Smallest block size must be large enough to hold a Block structure.
-  STATIC_ASSERT(smallest_word_size * sizeof(MetaWord) >= sizeof(Block));
+  // Block size must be exactly one word size.
+  STATIC_ASSERT(sizeof(Block) == BytesPerWord);
   STATIC_ASSERT(num_lists > 0);
 
 public:
 
-  // Minimal word size a block must have to be manageable by this structure.
-  const static size_t MinWordSize = smallest_word_size;
```

Comment:
Maybe keep MinWordSize = 1, then we can also keep using it, e.g. in code below:

int index = (int)(word_size - MinWordSize);

and many other places. Otherwise I wouldn't know what that '1' means.